src/utils/generate-question-states.ts

```diff
@@ -0,0 +1,57 @@
+import quizData from 'data/should-i-train.json'
+import capitalize from 'lodash/capitalize'
+import findIndex from 'lodash/findIndex'
+import toUpper from 'lodash/toUpper'
+
+const getStateName = (questionId: string) => `entering${capitalize(questionId)}`
+
+const getCurrentQuestionIndex = (questionId: string) =>
+  findIndex(quizData.questions, {id: questionId})
+
+export const getPrevQuestion = (questionId: string) => {
+  const currentIndex = getCurrentQuestionIndex(questionId)
+  return quizData.questions[currentIndex - 1]
+}
+
+export const getNextQuestion = (questionId: string) => {
+  const currentIndex = getCurrentQuestionIndex(questionId)
+  return quizData.questions[currentIndex + 1]
+}
+
+export interface Question {
+  value: string
+  id: string
+  byline?: string
+  options: {
+    value: string
+    score: number
+  }[]
+}
+
+const getActionsForQuestion = (questionId: string) => {
+  const prevQuestion = getPrevQuestion(questionId)
+  const nextQuestion = getNextQuestion(questionId)
+  const backAction = !!prevQuestion
+    ? {BACK: {target: getStateName(prevQuestion.id)}}
+    : {}
+  const confirmAction = !!nextQuestion
+    ? {
+        [`CONFIRM_${toUpper(questionId)}`]: {
+          target: getStateName(nextQuestion.id),
+          actions: ['assignScore'],
+        },
+      }
+    : {target: 'confirming', actions: ['assignScore']}
+
+  return {on: {...backAction, ...confirmAction}}
+}
+
+const getStateForQuestion = (question: Question) => {
+  const actions = getActionsForQuestion(question.id)
```

Comment:
I used these functions to generate the xstate states for me.

Reply:
nifty